qemu/tests/usb_redir.py

```diff
@@ -72,6 +84,35 @@ def _host_config_check():
                 err_msg = "Fail to install 'virt-viewer' on host, "
                 err_msg += "output: %s" % o
                 return (status, err_msg)
+        elif backend == 'tcp_socket':
+            usbredir_cmd = "rpm -q usbredir || yum install -y usbredir"
+            s, o = process.getstatusoutput(usbredir_cmd, shell=True)
+            if s:
+                status = False
+                err_msg = "Fail to install 'usbredir' on host, "
+                err_msg += "output: %s" % o
+                return (status, err_msg)
+            usbredirserver_query = "rpm -q usbredir-server"
+            s, o = process.getstatusoutput(usbredirserver_query, shell=True)
+            if s:
+                usbredir_query = "rpm -q usbredir"
+                usbredir_ver = process.getoutput(usbredir_query, shell=True)
+                (build, arch) = usbredir_ver.rsplit(".", 1)
+                brew_cmd = "brew download-build --arch=%s %s" % (arch, build)
+                s, o = process.getstatusoutput(brew_cmd, shell=True)
+                if s:
+                    status = False
+                    err_msg = "Fail to download build '%s', " % build
+                    err_msg += "output: %s" % o
+                    return (status, err_msg)
+                usbredirserver_cmd = "rpm -i usbredir-server*"
+                s, o = process.getstatusoutput(usbredirserver_cmd, shell=True)
+                if s:
+                    status = False
+                    err_msg = "Fail to install 'usbredir-server' on host, "
+                    err_msg += "output: %s" % o
+                    return (status, err_msg)
+            _start_usbredir_server()
```

Comment:
IMO, it is not belong to `_host_config_check`,  it is a step on the same level as `_start_spice_redirection`. so i suggest:
```
_host_config_check
_start_usbredir_server
_usbredir_preprocess
```

Reply:
Updated, thanks!